File: week_03/src/Question_19.js
// Author : Zaid Nayeb
// Date   : 2025/11/26
/*
19.	Pattern Printing – Pyramid Example: 
Input: 
5 
Output: 
    *   
   ***   
  *****   
 ******* 
********* 

*/

rightTriangle(10)
function rightTriangle(lines) {
    for (let i = 1; i < lines; i++) {
        for (let j = lines; j >= i; j--) {
            process.stdout.write("  ");
        }
        for (let k = 1; k <= i; k++) {
            process.stdout.write(" *");
        }
        for (let l = 0; l < i - 1; l++) {
            process.stdout.write(" *")

        }
        console.log();
    }
}

File: week_02/src/main/java/Question19.java
/*
Author = Zaid Nayeb
Date   = 18/11/2025

 19. Reverse a number
Description: Reverse the digits of an integer.
Example: Input:
1234 Output:
4321

*/

public class Question19 {
    public static void main(String[] args) {
        System.out.println(reverseNumber(-456));
    }

    public static long reverseNumber(long num){
        long reverse = 0;
        if (num < 0){
            num *= -1;
            while(num > 0){
                reverse += (num % 10) ;
                if (num >10) {
                    reverse *= 10;
                }
                num /= 10;
            }
            reverse *= -1;
            return reverse;

        }else{
            while(num > 0){
                reverse += (num % 10) ;
                if (num >10) {
                    reverse *= 10;
                }
                num /= 10;
            }
        return reverse;
        }
    }
}
